assets/js/googlesitekit/datastore/forms/index.test.js

```diff
@@ -19,73 +19,111 @@
+import { createTestRegistry } from 'tests/js/utils';
 import { STORE_NAME } from './constants';
 
 describe( 'core/forms store', () => {
-	const formName = 'Form 1';
-	const formData = {
-		key1: 'value1',
-		key2: 'value2',
-	};
-	const newFormName = 'Form 2';
-	const newFormData = {
-		key3: 'value3',
-		key4: 'value4',
-	};
 	let registry;
 
 	beforeEach( () => {
 		registry = createTestRegistry();
 	} );
 
-	afterEach( () => {
-		unsubscribeFromAll( registry );
-	} );
-
 	describe( 'actions', () => {
 		describe( 'setValues', () => {
 			it( 'requires the formName param', () => {
 				expect( () => {
 					registry.dispatch( STORE_NAME ).setValues();
 				} ).toThrow( 'formName is required for setting values.' );
 			} );
+
 			it( 'requires the formData param', () => {
+				const formName = 'Form 1';
+
 				expect( () => {
 					registry.dispatch( STORE_NAME ).setValues( formName );
 				} ).toThrow( 'formData must be an object.' );
 			} );
+
+			it( 'requires the formData param to be an object not an array', () => {
+				const formName = 'Form 1';
+
+				expect( () => {
+					registry.dispatch( STORE_NAME ).setValues( formName, [] );
+				} ).toThrow( 'formData must be an object.' );
+			} );
+
+			it( 'requires the formData param to be an object not a string', () => {
+				const formName = 'Form 1';
+
+				expect( () => {
+					registry.dispatch( STORE_NAME ).setValues( formName, 'formData' );
+				} ).toThrow( 'formData must be an object.' );
+			} );
 		} );
 	} );
 
 	describe( 'selectors', () => {
 		describe( 'getValue', () => {
-			it( 'works with a formName not previously assigned', async () => {
-				await registry.dispatch( STORE_NAME ).setValues( formName, { ...formData } );
+			it( 'works with a formName that does not exist', () => {
+				const formName = 'Form 1';
+				const formData = {
+					key1: 'value1',
+					key2: 'value2',
+				};
+				const newFormName = 'Form 2';
+
+				registry.dispatch( STORE_NAME ).setValues( formName, { ...formData } );
+
+				const form = registry.select( STORE_NAME ).getValue( newFormName, 'key2' );
+				expect( form ).toEqual( undefined );
+			} );
+
+			it( 'works with formData where the key does not exist', () => {
+				const formName = 'Form 1';
+				const formData = {
+					key1: 'value1',
+					key2: 'value2',
+				};
+
+				registry.dispatch( STORE_NAME ).setValues( formName, { ...formData } );
+
+				const form = registry.select( STORE_NAME ).getValue( formName, 'key3' );
+				expect( form ).toEqual( undefined );
+			} );
+
+			it( 'works with an existing formName', () => {
+				const formName = 'Form 1';
+				const formData = {
+					key1: 'value1',
+					key2: 'value2',
+				};
+
+				registry.dispatch( STORE_NAME ).setValues( formName, { ...formData } );
 
 				const { key2 } = formData;
-				const form = registry.select( STORE_NAME ).getValue( formName, Object.keys( formData )[ 1 ] );
+				const form = registry.select( STORE_NAME ).getValue( formName, 'key2' );
```

Comment:
`form` is not an accurate name for the returned value since you're getting a single value by key from the form.

The destructured `key2` is also a bit unnecessary.

Can we add a similar test for `setValues`? We should set values with one set, and then set them again leaving at least one of the original keys untouched, then assert that the untouched key and updated keys return their expected values.

Reply:
@aaemnnosttv I've added a new test for updating keys